anymail/backends/mailtrap.py

```diff
@@ -0,0 +1,248 @@
+
+
+class EmailBackend(AnymailRequestsBackend):
+    """
+    Mailtrap API Email Backend
+    """
+
+    esp_name = "Mailtrap"
+
+    def __init__(self, **kwargs):
+        """Init options from Django settings"""
+        self.api_token = get_anymail_setting(
+            "api_token", esp_name=self.esp_name, kwargs=kwargs, allow_bare=True
+        )
+        api_url = get_anymail_setting(
+            "api_url",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default="https://send.api.mailtrap.io/api/",
+        )
+        if not api_url.endswith("/"):
+            api_url += "/"
+
+        test_api_url = get_anymail_setting(
+            "test_api_url",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default="https://sandbox.api.mailtrap.io/api/",
+        )
+        if not test_api_url.endswith("/"):
+            test_api_url += "/"
+        self.test_api_url = test_api_url
+
+        bulk_api_url = get_anymail_setting(
+            "bulk_api_url",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default="https://bulk.api.mailtrap.io/api/",
+        )
+        if not bulk_api_url.endswith("/"):
+            bulk_api_url += "/"
+        self.bulk_api_url = bulk_api_url
+
+        self.test_inbox_id = get_anymail_setting(
+            "test_inbox_id",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+        )
+
+        self.testing_enabled = get_anymail_setting(
+            "testing",
+            esp_name=self.esp_name,
+            kwargs=kwargs,
+            default=False,
+        )
+
+        if self.testing_enabled:
+            if not self.test_inbox_id:
+                warnings.warn(
+                    "Mailtrap testing is enabled, but no test_inbox_id is set. "
+                    "You must set test_inbox_id for Mailtrap testing to work.",
+                    AnymailWarning,
+                )
+            api_url = self.test_api_url
+            self.bulk_api_url = self.test_api_url
+
+        super().__init__(api_url, **kwargs)
+
+    def build_message_payload(self, message, defaults):
+        return MailtrapPayload(message, defaults, self)
+
+    def parse_recipient_status(
+        self, response, payload: MailtrapPayload, message: AnymailMessage
+    ):
+        parsed_response = self.deserialize_json_response(response, payload, message)
+
+        if (
+            not parsed_response.get("success")
+            or ("errors" in parsed_response and parsed_response["errors"])
+            or ("message_ids" not in parsed_response)
+        ):
+            raise AnymailRequestsAPIError(
+                email_message=message, payload=payload, response=response, backend=self
+            )
+
+        # Not the best status reporting. Mailtrap only says that the order of message-ids will be in this order (but JSON is unordered?)
+        recipient_status_order = [*message.to, *message.cc, *message.bcc]
```

Comment:
Mailtrap's status reporting is pretty poor, and I'm not even sure if this will work consistently, given that it is depending on ORDERED JSON responses :facepalm:

Reply:
Will do. Do you think there would be any value in sorting the recipients by email address before sending?

Reply:
I would not try to sort recipients. I don't think it would help anything, and there might be cases where senders are deliberately putting addresses in a specific order.